Source/WebCore/dom/Node.h

```diff
@@ -777,7 +777,7 @@ class Node : public EventTarget {
     mutable OptionSet<StateFlag> m_stateFlags;
 
     ContainerNode* m_parentNode { nullptr };
-    CheckedPtr<TreeScope> m_treeScope;
+    TreeScope* m_treeScope { nullptr };
```

Comment:
Could this be a WeakPtr?

Reply:
I think we should do that in a separate patch.

Reply:
Right, I intentionally went back to raw pointers in hot code paths to limit perf risks. We should definitely try in a follow-up after A/B testing though.

Reply:
Then yes, every raw pointer I introduced in this PR should indeed be a WeakPtr. We just need to A/B test because DOM code is hot.